test/unit/auth_sources/auth_source_ldap_test.rb

```diff
@@ -258,6 +258,27 @@ def setup
     end
   end
 
+  context 'save external avatar' do
+    def teardown
+      FileUtils.rm_rf("#{Rails.public_path}/assets/avatars")
+    end
+
+    test 'store_avatar can save 8bit ascii files' do
+      auth = AuthSourceLdap.new
+      file = Tempfile.new('foo')
```

Comment:
I know we interact with the file system on other instances, such as the Puppet parser tests, seeds tests, etc... Can we do this in memory instead? Two issues I see with this are:

* It's slow due to the file system 
* rm_rf any file can be dangerous (on the teardown block). It'd also delete avatars that already exist?

I thought about stubbing things but that's not very easy to do with files & it ties the test too much with the implementation of `store_avatar`. 

How about doing as they suggest [in this article](https://mojolingo.com/blog/2013/tests-mocks-rspec/)? You can just add 'fakefs' to bundler/test, no need to package anything and we'd solve the issue here and lay the ground to get rid of other slow tests. Last time I measured it, I think `test/lib/tasks/seeds_test.rb` was the slowest test file because of the file system.